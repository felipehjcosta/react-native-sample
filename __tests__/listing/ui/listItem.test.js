import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ReactTestRenderer from 'react-test-renderer'
import ListItem from '../../../src/listing/ui/listItem'

Enzyme.configure({adapter: new Adapter()})

describe('LoadingItem Component', () => {
  it('renders correctly', () => {
    const item = {
      img_url: 'https://imgs.nestimg.com/casa_300_m2_108811252799247473.jpg',
      price_formatted: 'R$ 750.000',
      title: 'Campo Grande,Rio de Janeiro,Rio De Janeiro',
      lister_url: 'https://www.nestoria.com.br/detail/0000000108550797542396092/title/5/1-1?serpUid=&pt=1&ot=1&l=rio-de-janeiro&did=41_default&utm_source=api&utm_medium=external'
    }
    const wrapper = shallow(<ListItem item={item} />)

    const tree = ReactTestRenderer.create(wrapper).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
