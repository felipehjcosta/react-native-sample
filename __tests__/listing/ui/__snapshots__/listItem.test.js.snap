// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingItem Component renders correctly 1`] = `
<View
  stye={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
    style={
      Array [
        Array [
          Object {
            "height": 400.2,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 8,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          },
        ],
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "marginBottom": 8,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 8,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#48BBEC",
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      R$ 750.000
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#656565",
          "fontSize": 20,
        }
      }
    >
      Campo Grande,Rio de Janeiro,Rio De Janeiro
    </Text>
  </View>
</View>
`;
